mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -266,6 +272,68 @@ class ConvConverter final : public OpConversionPattern<tosa::Conv2DOp> {
   }
 };
 
+class Conv3DConverter final : public OpConversionPattern<tosa::Conv3DOp> {
```

Comment:
Begging for abstraction.  Conv3DConverter is nearly identical to ConvConverter, except for the op type and the sizes of biasShape and reassociations.